tests/test_manifest.py

```diff
@@ -301,30 +301,41 @@ def test_get_bool_value(self, manifest, value, expected):
         manifest.contents['test'] = value
         assert manifest.get_bool_value('test') is expected
 
-    @mock.patch('ship_it.manifest.Manifest.get_config_args_and_flags',
-                return_value=(['cfg arg'], [('cfg', 'flag')]))
-    @mock.patch('ship_it.manifest.Manifest.get_single_flags',
-                return_value=[('single', 'flag')])
-    @mock.patch('ship_it.manifest.Manifest.get_dependency_flags',
-                return_value=[('depends', 'weird-dependency == 0.1')])
-    @mock.patch('ship_it.manifest.Manifest.get_exclude_flags',
-                return_value=[('exclude', '**.pyc'), ('exclude', '**.pyo')])
+    @pytest.mark.parametrize('cfg_update,expected_user,expected_group', [
+        ({},                'ship_it', 'ship_it'),
+        ({'user': 'root'},  'root',    'ship_it'),
+        ({'group': 'root'}, 'ship_it', 'root'),
+        ({'user': 'root',
+          'group': 'root'}, 'root', 'root'),
+    ])
+    @mock.patch('ship_it.manifest.Manifest.get_config_args_and_flags')
+    @mock.patch('ship_it.manifest.Manifest.get_single_flags')
+    @mock.patch('ship_it.manifest.Manifest.get_dependency_flags')
+    @mock.patch('ship_it.manifest.Manifest.get_exclude_flags')
     def test_get_overall_args(self, mock_excludes, mock_depends, mock_single,
-                              mock_cfg, manifest):
+                              mock_cfg, manifest, cfg_update, expected_user, expected_group):
+        # Return values must be set here, rather than in the decorator,
+        # to reset their state for each iteration.
```

Comment:
actual_flags will have two copies of every flag on the 2nd iteration, three copies on the third iteration, etc. Took me about four times as long to figure out why that was happening as it took to make the original change, but it forced me to learn a lot more about pytest than I did when I started. :-)